### PR TITLE
New API for reading kubectl file

### PR DESCRIPTION
**Pull request by scrosby**
## Changes proposed in this PR

- Switch to using a different API to parse .kube/config files. 

## Why are we making these changes?

Make it easier to directly decide which context out of the config we should be running.

### CODE REVIEW

File: scheduler/src/cook/kubernetes/compute_cluster.clj [hunk 1]
```diff
@@ -23,11 +23,12 @@
...
            (io.kubernetes.client.openapi.models V1Node V1Pod)
-           (io.kubernetes.client.util Config)
-           (java.io File FileInputStream)
+           (io.kubernetes.client.util Config KubeConfig)
+           (java.io File FileInputStream InputStreamReader)
            (java.util.concurrent Executors ExecutorService ScheduledExecutorService TimeUnit)
            (java.util UUID)
-           (okhttp3 OkHttpClient$Builder)))
+           (okhttp3 OkHttpClient$Builder)
+           (java.nio.charset StandardCharsets)))
```
Comment (dposada) [inline scheduler/src/cook/kubernetes/compute_cluster.clj:31]: Lexicographic

File: scheduler/src/cook/kubernetes/compute_cluster.clj [hunk 2]
```diff
@@ -545,9 +546,16 @@
...
-                     (Config/fromConfig config-file)
-                     (ApiClient.))
+  (log/info "API Client config file" config-file)
+  (let [^ApiClient api-client (if (some? config-file)
+                                (let [^KubeConfig kubeconfig
+                                      (-> config-file
+                                          (FileInputStream.)
+                                          (InputStreamReader. (.name (StandardCharsets/UTF_8)))
+                                          (KubeConfig/loadKubeConfig))]
+                                  (.setFile kubeconfig (File. config-file)) ; Workaround client library bug. Needed to avoid a NPE. Use an absolute path so this isn't used.
```
Comment (dposada) [inline scheduler/src/cook/kubernetes/compute_cluster.clj:556]: The comment isn't clear to me -- what was the bug / NPE?

File: scheduler/src/cook/kubernetes/compute_cluster.clj (hunk 1 above)
Comment (scrosby) [reply scheduler/src/cook/kubernetes/compute_cluster.clj]: Fixed.

File: scheduler/src/cook/kubernetes/compute_cluster.clj (hunk 2 above)
Comment (scrosby) [reply scheduler/src/cook/kubernetes/compute_cluster.clj]: Rewrote

**Pull request merged**